# Mecánica de fluidos - Práctico 1
__UNLZ - FI -- Jupyter Notebook + Python 3__
   
__Alumno: Lorenzo, Daniel Antonio__

## Problema 1
Con referencia  a la figura y despreciando el rozamiento entre el pistón A y el cilindro que contiene el gas, determinar la presión manométrica en B en metros de columna de agua. Suponer el peso específico del gas 0,56 kg/m³.

![tp1-01.png](tp1-01.png)

### Solución
El área de $A = A_a = \pi r^2 = \pi \, (30 \, m)^2 $

In [3]:
import math
r = 60 / 2 # m
A_a = math.pi * r**2 # m^2
print('A_a =',round(A_a,2),'m^2')

A_a = 2827.43 m^2


La presión relativa bajo A es
$$ p_a = \frac{P_A}{A_a} $$

In [5]:
P_A = 1600 * 1000 # kgf peso del pistón
p_a = P_A / A_a # presión relativa bajo A
print('p_a =',round(p_a,2),'kgf/m^2')

p_a = 565.88 kgf/m^2


Para un gas ideal en condiciones isotérmicas
$$ p v = p_1 v_1 = R \qquad \qquad [1] $$
ahora como
$$ \gamma = g \delta \quad \Rightarrow \frac{1}{\delta} = \frac{g}{\delta} $$ y como 
$$ v = \frac{1}{\delta} \quad \Rightarrow \quad v = \frac{q}{\gamma} \qquad [2] $$
reemplazando [2] en [1] resulta:
$$ p \frac{g}{\gamma} = p_1 \frac{g_1}{\gamma_1} = R \qquad [3] $$
Suponiendo que el intervalo de elevación no es lo suficientemente grande como para que introduzca una variación significativa en $g$, de manera que $g$ puede considerarse como una constante. Luego, dividiendo por $g$ en [3]
$$ \frac{g}{\gamma} = \frac{p_1}{\gamma_1} = \frac{R}{g} = C \qquad [4] \qquad \therefore \quad \gamma = \frac{P}{C} \qquad [5] $$
Utilizando [5], puede expresarse la ecuación diferencial básica como sigue:
$$ \frac{dP}{dz} = \gamma = \frac{P}{C} $$
Separando variables e integrando entre $p_1$ y $p$ y entre $z_1$ y $z$, se tiene
## $$ \int_{p_1}^p \frac{dP}{P} = \int_{z_1}^z \frac{dz}{C} \qquad \Rightarrow \qquad \left. \ln p \right|_{p_1}^p = \left. \frac{z}{C} \right|_{z_1}^z \qquad \Rightarrow \qquad \ln \frac{p}{p_1} = \frac{1}{C} (z - z_1) $$
despejando $p$ y utilizando la ecuación [4] $p_1/ \gamma_1 = C$
## $$ \frac{p}{p_1} = e^{(\gamma_1 / p_1) (z-z_1)} $$
## $$ p = p_1 \, e^{(\gamma_1 / p_1) (z-z_1)} \qquad [7] $$
Como $p_1 = p_a = 565,88 \, kgf/m^2$ y $z_1 = 0$, $z = 70 \, m$ reemplazando en [7] resulta

In [10]:
p_1 = p_a
gamma = 0.56 # kgf/m^3
z = 70
z_1 = 0 # m
p_b = p_1 * math.e**(gamma/p_1*(z-z_1))
print('p_b =',round(p_b,2),'kgf/m^2')

p_b = 606.47 kgf/m^2


### $$ p_{H_2O} = \gamma_{H_2O} \cdot h = 1000 \, kgf/m^3 \cdot 1 \, m = 1000 \, kgf/m^2 = 1 \, mca $$
## $$ p_B = 606,47 \, kgf/m^2 = 0,606 \, mca = 60,64 \, cmca $$

In [1]:
%reset -f

## Problema 2
El aire del recipiente de la izquierda de la figura está a una presión de -23 cm de columna de mercurio. Determinar la cota del líquido manométrico en la parte derecha, en A.

![tp1-02.png](tp1-02.png)

### Solución
Considero que la presión dada para el aire en el recipiente de la izquierda es la presión a los 36m. Luego la presión en B es igual a la presión en A porque están al mismo nivel, (horizontal) y están interconectados por el mismo fluido estático.

    1 cmHg = 135,95 kgf/m²       1 kgf/cm² = 10000 kgf/m²

In [1]:
from sympy import *

In [4]:
z = symbols('z')
PB_aire = 135.95 * -23 # kgf/m^2
gamma_aceite = 0.8 * 1000 # kgf/m^3
h_aceite = 4 # m
gamma_l = 1.6 * 1000 # kgf/m^3
PA_aire = 2000 # kgf/m^3
gamma_agua = 1000 # kgf/m^3
h_agua = 1.5 # m
# Presión en A
PA = PA_aire  + gamma_agua*(h_agua + z) 
# Presión en B
PB = PB_aire + gamma_aceite*h_aceite + gamma_l*z  
# ecuación 1: PA = PB
eq1 = Eq(PA , PB)
# Resuelve ecuación 1, en z
seq1 = solve(eq1,z)
z = seq1[0]
print('z =',round(z,4),'m') 

z = 5.7114 m


In [8]:
h_A = 32 - z
print('Altura en A:')
print('h_A =',round(h_A,1),'m')

Altura en A:
h_A = 26.3 m


In [ ]:
%reset -f

## Problema 3
Un tanque cilíndrico contiene agua hasta una altura de 50 mm. Dentro de éste se encuentra un tanque cilíndrico más pequeño, abierto, que contiene queroseno hasta una altura h, con una densidad de 0,8. Las siguientes presiones se conocen en los manómetros indicados: $p_B = 13,80 \, kPa$, $p_C = 13,82 \, kPa$. ¿Cuáles son la presión manométrica $p_A$ y la altura $h$ del queroseno? Suponga que se impide el movimiento del queroseno hacia la parte superior del tanque.
![tp1-03.png](tp1-03.png)

### Solución
$
\left\{ \begin{array}{l}
\mathrm{ 1 \, kgf/cm^2 = 98066,52 \, Pa = 98,07 \, kPa } \\
\mathrm{ 1 \, kgf/m^2 = 9,81 \, Pa = 9,81 \times 10^{-3} \, kPa}
\end{array}
\right.
$

### $$ p_A = p_C - \gamma_{H_2O} \cdot h_{H_2O} $$

In [4]:
g = 9.81 # m/s^2
p_C = (13.82*10**3) * (1/g) # kgf/m^2
gamma_agua = 1000 # kgf/m^3
h_agua = 50 / 1000 # m
p_A = p_C - gamma_agua * h_agua
print('p_A =',round(p_A,2),'kgf/m^2',' = ',round(p_A/98.07,2),'kPa')

p_A = 1358.77 kgf/m^2  =  13.86 kPa


__Cálculo de h__

$
\begin{array}{l}
p_C - \gamma_{H_2O} \cdot h = p_B - \gamma_k \cdot h \\
p_C - p_B = (\gamma_{H_2O} - \gamma_k) \cdot h 
\end{array}
$
## $$ \Rightarrow \; h = \frac{p_C-p_B}{\gamma_{H_2O}-\gamma_k} $$

In [5]:
p_B = (13.8*10**3) * (1/g) # kgf/m^2
gamma_k = 0.8 * 1000 # kgf/m^3

h = (p_C - p_B)/(gamma_agua - gamma_k)
print('h =',round(h,5),'m')

h = 0.01019 m


In [ ]:
%reset -f

## Ejercicio 4
¿Cuál es la presión manométrica dentro del tanque de aire?
![tp1-04.png](tp1-04.png)

### Solución
Considero:

$
\begin{array}{l}
P_a = P_b \\
P_b = P_c \\
P_d = P_c - \gamma_{Hg} \cdot 200 \, mm \\
P_e = P_d + \gamma_{H_2O} \cdot 100 \, mm \\
P_f = P_e \\
P_f = \gamma_{Hg} 260 mm
\end{array}
$

In [2]:
# Cálculo de presión en f y e
gamma_Hg = 13600 # kgf/m^3
P_f = gamma_Hg * (460-200)/1000
P_e = P_f
print('P_f =',P_f,'kgf/m^2')
print('P_e =',P_e,'kgf/m^2')

P_f = 3536.0 kgf/m^2
P_e = 3536.0 kgf/m^2


In [3]:
# Cálculo de presión en d
gamma_agua = 1000 # kgf/m^2
P_d = P_e - gamma_agua * 100/1000
print('P_d =',P_d,'kgf/m^2')

P_d = 3436.0 kgf/m^2


In [6]:
# Cálculo de presión en c, b y a
P_c = P_d + gamma_Hg * 200/1000
P_b = P_c
P_a = P_b
print('P_c =',P_c,'kgf/m^2')
print('P_b =',P_b,'kgf/m^2')
print('P_a =',P_a,'kgf/m^2')

P_c = 6156.0 kgf/m^2
P_b = 6156.0 kgf/m^2
P_a = 6156.0 kgf/m^2


In [7]:
%reset -f

## Ejercicio 5
Encuentre la diferencia de presión entre los tanques A y B si $d_1 = 300 \, mm$, $d_2 = 150 \, mm$, $d_3 = 460 \, mm$, $d_4 = 200 \, mm$ y la densidad relativa del mercurio es 13,6.
![tp1-05.png](tp1-05.png)

### Solución
Considero $\sin 45^\circ = h/4$

$h = \sin 45^\circ d_4$
$$ P_a = P_b + \gamma_{Hg} \cdot (d_4 \cdot \sin 45^\circ + d_3) - \gamma_{H_2O} \cdot d_1 $$
### $$ \Delta P_{ab} = P_a - P_b = \gamma_{Hg} \cdot (d_4 \cdot \sin 45^\circ + d_3) - \gamma_{H_2O} \cdot d_1 $$

In [10]:
import math
gamma_Hg = 13600 # kgf/m^3
gamma_agua = 1000 # kgf/m^3
d1 = 300/1000 # m
d2 = 150/1000 # m
d3 = 460/1000 # m
d4 = 200/1000 # m
ang = 45 * math.pi/180 # radianes
DP_ab = gamma_Hg * (d4*math.sin(ang)+d3)-gamma_agua*d1
print('DP_ab =',round(DP_ab,2),'kgf/m^2')

DP_ab = 7879.33 kgf/m^2


In [1]:
%reset -f

## Ejercicio 6
Un elevador hidráulico está provisto de dos émbolos, uno de superficie igual a 1 m² y otro de superficie igual a 5 cm². ¿Qué fuerza mínima debemos ejercer para elevar una masa de 1000 kg? ¿En cuál de los émbolos?
![tp1-06.png](tp1-06.png)

### Solución
$
\begin{array}{l}
A_1 = 5 \, cm^2 = 5 \times 10^{-4} \, m^2 \\
A_2 = 1 \, m^2
\end{array}
$
### $$ \frac{F_1}{A_1} = \frac{F_2}{A_2} $$

In [4]:
A1 = 5*10**(-4) # m^2
A2 = 1 # m^2
m = 1000 # kg
g = 9.81 # m/s^2 (aceleración de la gravedad)
F2 = m*g # N
P2 = F2/A2 # N/m^2
P1 = P2
F1 = P2*A1 # N
print('P1 = P2 =',P1,'N/m^2')
print('F1 =',F1,'N, (fuerza mínima)')

P1 = P2 = 9810.0 N/m^2
F1 = 4.905 N, (fuerza mínima)


La fuerza debe aplicarse en el émbolo más pequeño ya que la relación entre las fuerzas viene dada por:
### $$ \frac{F_1}{F_2} = \frac{A_1}{A_2} $$
En esta forma, con $A_B > A_A$ puede obtenerse una considerable ventaja mecánica.

In [3]:
%reset -f

## Ejercico 7
¿Cuál es la presión $P_A$? La densidad realtiva del aceite es 0,8
![tp1-07.png](tp1-07.png)

### Solución
$$ P_A = \gamma_{Hg} \cdot (0,3 m) - \gamma_{H_2O} \cdot (1,6 m) - (0,8) \cdot (\gamma_{H_2O}) \cdot (3 m) $$

In [2]:
gamma_Hg = 13600 # kgf/m^3
gamma_agua = 1000 # kgf/m^3
gamma_aceite = 0.8*gamma_agua
P_A = gamma_Hg*0.3 - gamma_agua*(4.6-3) - gamma_aceite*3
print('P_A =',round(P_A,2),'kgf/m^2')

P_A = 80.0 kgf/m^2


In [ ]:
%reset -f

## Ejercicio 8
Los compartimentos B y C de la figura están cerrados y llenos de aire. La presión atmosférica es de 1 kg/cm². Cuando los manómetros A y D marcan las lecturas indicadas, ¿Qué valor tendrá la cota x en el manómetro de mercurio E?
![tp1-08](tp1-08.png)

### Solución
$$ P_A = 21000 \, kgf/m^2 $$
$$ P_C = - \gamma_{Hg} \cdot (0,25 m) $$

$$ P_A = P_C + \gamma_{Hg} \cdot x \qquad [1] $$

### $$ x = \frac{P_A + \gamma_{Hg} \cdot (0,25 m)}{\gamma_{Hg}} $$

In [5]:
# Con sympy (solamente para probar la biblioteca o library)
# es una ecuación una ecuación de 1 incógnita, como que no hace falta
from sympy import *
x = symbols('x')
P_A = 2.1 * 10000 # kgf/m^2 (presión en A)
gamma_Hg = 13600 # kgf/m^3
P_C = -gamma_Hg * 0.25 # kgf/m^2 (presión en C)
eq1 = Eq(P_A , P_C + gamma_Hg * x) # ecuación 1
seq1 = solve(eq1,x) # resuelve la ecuación [1]
x = seq1[0]
print('x =',round(x,4),'m')

x = 1.7941 m


In [6]:
x = (P_A + gamma_Hg * 0.25)/gamma_Hg
print('x =',round(x,4),'m')

x = 1.7941 m


In [7]:
%reset -f

## Ejercicio 9
Encuentre la distancia d para el tubo en U.
![tp1-09.png](tp1-09.png)

### Solución
$$P_B = P_A + \gamma_{ac} (0,05m+0,01m)$$
$$P_C = P_B - \gamma_{Hg} (0,02m) $$
$$P_D = P_C - 1,4 \gamma_{H2O} \cdot d $$
### $$ P_D = P_A + (0,6 \cdot \gamma_{H_2O}) (0,06m)-\gamma_{Hg}(0,02m)- (1,4 \cdot \gamma_{H_2O}) \cdot d $$
Como $P_D = 0$ resulta
### $$ d = \frac{P_A + (0,6 \cdot \gamma_{H_2O})(0,6m) - \gamma_{Hg}(0,02m)}{1,4 \cdot \gamma_{H_2O}} $$

In [9]:
# 1 kPa = 101.97 kgf/m^2
P_A = 2.7 * 101.97 # kgf/m^2
gamma_Hg = 13600 # kgf/m^3
gamma_agua = 1000 # kgf/m^3
gamma_aceite = 0.6*gamma_agua # kgf/m^3
gamma_liq = 1.4*gamma_agua # kgf/m^3

d = (P_A + gamma_aceite*0.06 - gamma_Hg*0.02)/gamma_liq
print('d =',round(d,3),'m =',round(d*1000,3),'mm')

d = 0.028 m = 28.085 mm


In [ ]:
%reset -f

## Ejercicio 10
El depósito de la figura contiene agua.
1. Si abrimos la llave de paso, ¿qué altura tendrá el agua en cada lado del depósito cuando se alcance el equilibrio?
2. ¿Qué cantidad de agua pasará de un recipiene al otro hasta que se alcance el equilibrio?
![tp1-10.png](tp1-10.png)

### Solución
1) En el equilibrio la presión en el fondo de los recipientes es igual, por lo tanto:

$h_{iA} =$ altura inicial en A, $h_{iB} =$ altura inicial en B, $h_{fA} =$ altura final en A, $h_{fB} =$ altura final en B, $A_A =$ área A, $A_B =$ área B

### $$ \gamma_{agua} \cdot h_{fA} = \gamma_{agua} \cdot h_{fB} + P_{atm} \quad \Rightarrow \quad h_{fA} = h_{fB} + \frac{P_{atm}}{\gamma_{agua}} \qquad [1] $$
además el volumen total de agua no cambia por lo tanto:
### $$ h_{iA} \cdot Area_A + h_{iB} \cdot Area_B = h_{fA} \cdot Area_A + h_{fB} \cdot Area_B \qquad [2] $$

In [38]:
from sympy import *
hf_A, hf_B = symbols('hf_A , hf_B')
hi_A = 15 # m altura inicial A
hi_B = 10 # m altura inicial B
Area_A = 10 # m^2
Area_B = 20 # m^2
gamma_agua = 1000 # kgf/m^3
P_atm = 10000 # kgf/m^2
eq1=Eq(hf_A , hf_B + P_atm/gamma_agua) # ecuación 1
eq2=Eq(hi_A*Area_A+hi_B*Area_B , hf_A*Area_A+hf_B*Area_B) # ecuación 2
Sol=solve([eq1,eq2],[hf_A,hf_B],dict=True)
print(Sol)
hf_A = Sol[0][hf_A] # altura final A
hf_B = Sol[0][hf_B] # altura final B
print('Entonces:')
print('hf_A =',round(hf_A,3),'m')
print('hf_B =',round(hf_B,3),'m')

[{hf_A: 18.3333333333333, hf_B: 8.33333333333333}]
Entonces:
hf_A = 18.333 m
hf_B = 8.333 m


2) La cantidad de agua que pasa de un recipiente al otro será:
### $$ V_{fA} - V_{iA} = h_{fA} \cdot Area_A - h_{iA} \cdot Area_A $$
## $$ \Delta V_A = -\Delta V_B $$

In [40]:
DVA = hf_A * Area_A -hi_A * Area_A
print('DVA =',round(DVA,3),'m^3')

DVA = 33.333 m^3


In [ ]:
%reset -f